packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -1639,46 +1803,45 @@ class ObjectsInRedisClient {
             if (err) {
                 return tools.maybeCallbackWithError(callback, err);
             } else {
-                return this._touch(id, name, options, callback, meta);
+                return this._touch(id, name, callback, meta);
             }
         });
     }
 
-    touchAsync(id, name, options) {
-        return /** @type {Promise<void>} */ (
-            new Promise((resolve, reject) => this.touch(id, name, options, err => (err ? reject(err) : resolve())))
-        );
+    touchAsync(id: string, name: string, options: CallOptions): Promise<void> {
+        return new Promise((resolve, reject) => this.touch(id, name, options, err => (err ? reject(err) : resolve())));
     }
 
-    async _rmHelper(keys, callback) {
-        if (!keys || !keys.length) {
-            return tools.maybeCallback(callback);
+    private async _rmHelper(keys: string[]): Promise<void> {
+        if (!keys.length) {
+            return;
         } else {
             if (!this.client) {
-                return tools.maybeCallbackWithError(callback, ERRORS.ERROR_DB_CLOSED);
+                throw new Error(ERRORS.ERROR_DB_CLOSED);
             }
 
             for (const id of keys) {
-                try {
-                    await this._delBinaryState(id.replace(/\$%\$meta$/, '$%$data'));
-                    await this.client.del(id);
-                } catch (e) {
-                    return tools.maybeCallbackWithRedisError(callback, e);
-                }
+                await this._delBinaryState(id.replace(/\$%\$meta$/, '$%$data'));
+                await this.client.del(id);
             }
-            return tools.maybeCallback(callback);
         }
     }
 
-    async _rm(id, name, options, callback, meta) {
+    private async _rm(
+        id: string,
+        name: string,
+        options: CallOptions,
+        meta?: any
+    ): Promise<ioBroker.RmResult[] | undefined> {
         if (meta && !meta.isDir) {
             // it is file
             const metaID = this.getFileId(id, name, true);
             const dataID = this.getFileId(id, name, false);
-            this.delObject(dataID, _err => this.delObject(metaID, callback));
+            await this.delObjectAsync(dataID);
```

Comment:
Not the same logic because old code ignored error on data delete and still deleted meta ... now both would stay .... and callback not called or?

Reply:
ahh ok, callback is gone here ... ok